week06_policy_based/atari_wrappers.py

```diff
@@ -108,31 +108,33 @@ def reset(self, **kwargs):
 class ImagePreprocessing(ObservationWrapper):
     """Preprocesses image-observations by possibly grayscaling and resizing."""
 
-    def __init__(self, env, width=84, height=84, grayscale=True):
+    def __init__(self, env, height=84, width=84, grayscale=True):
         super().__init__(env)
-        self.width = width
         self.height = height
+        self.width = width
         self.grayscale = grayscale
         ospace = self.env.observation_space
         low, high, dtype = ospace.low.min(), ospace.high.max(), ospace.dtype
         if self.grayscale:
             self.observation_space = Box(
                 low=low,
                 high=high,
-                shape=(width, height),
+                shape=(height, width),
                 dtype=dtype,
             )
         else:
-            obs_shape = (width, height) + self.observation_space.shape[2:]
             self.observation_space = Box(
-                low=low, high=high, shape=obs_shape, dtype=dtype
+                low=low,
+                high=high,
+                shape=(height, width, *self.observation_space.shape[2:]),
+                dtype=dtype,
             )
 
     def observation(self, observation):
         """Performs image preprocessing."""
         if self.grayscale:
             observation = cv2.cvtColor(observation, cv2.COLOR_RGB2GRAY)
-        observation = cv2.resize(observation, (self.width, self.height), cv2.INTER_AREA)
+        observation = cv2.resize(observation, (self.height, self.width), cv2.INTER_AREA)
```

Comment:
I think this is wrong -- the correct order is [(width, height)](https://docs.opencv.org/4.x/da/d6e/tutorial_py_geometric_transformations.html). Confusing, I know :(

(To make things more confusing, torchvision [uses (height, width)](https://pytorch.org/vision/main/generated/torchvision.transforms.functional.resize.html) by default.)

Reply:
A little bit misleading :)
Now it's fixed.